main.go

```diff
@@ -0,0 +1,121 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+
+	awsacm "github.com/aws/aws-sdk-go-v2/service/acm"
+	"github.com/skpr/certificate-expiry-notification-lambda/internal/acm"
+	"github.com/skpr/certificate-expiry-notification-lambda/internal/slack"
+	util "github.com/skpr/certificate-expiry-notification-lambda/internal/utils"
+)
+
+var (
+	today        = time.Now().UTC()
+	expiryDays   = getExpiryDays()
+	expiryWindow = today.AddDate(0, 0, expiryDays)
+)
+
+func getExpiryDays() int {
+	if os.Getenv("EXPIRY_DAYS") == "" {
+		return 45
+	}
+	days, err := strconv.Atoi(os.Getenv("EXPIRY_DAYS"))
+	if err != nil {
+		return 45
+	}
+
+	return days
+}
+
+func main() {
+	lambda.Start(lambdaHandler)
+}
+
+func lambdaHandler(ctx context.Context, rawEvent json.RawMessage) (map[string]interface{}, error) {
+	var event map[string]interface{}
+	err := json.Unmarshal([]byte(rawEvent), &event)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal event: %w", err)
+	}
+
+	config, err := util.LoadConfig(".")
+	if err != nil {
+		return nil, fmt.Errorf("failed to load config: %w", err)
+	}
+
+	errs := config.Validate()
+	if len(errs) > 0 {
+		return nil, fmt.Errorf("configuration error: %s", strings.Join(errs, "\n"))
+	}
+	slackClient, err := slack.NewClient(config.SlackWebhookURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Slack client: %w", err)
+	}
+
+	sess, err := awsconfig.LoadDefaultConfig(ctx)
+
+	if err != nil {
+		fmt.Println("Couldn't load configuration. Have you set up your AWS account?")
+		fmt.Println(err)
+		return nil, err
+	}
+
+	acmClient := awsacm.NewFromConfig(sess)
+
+	if event["detail-type"] == "ACM Certificate Approaching Expiration" && len(event["resources"].([]interface{})) > 0 {
```

Comment:
* Use a struct instead of the map[string]interface
* I would make the detail type value a const, check for it and use defensive programming to return the error early.